src/workers/newNotificationV2Mail.ts

```diff
@@ -448,32 +470,51 @@ const notificationToTemplateData: Record<NotificationType, TemplateDataFunc> = {
     return null;
   },
   post_mention: async (con, user, notification) => {
-    const post = await con.getRepository(SharePost).findOne({
+    const post = await con.getRepository(Post).findOne({
       where: { id: notification.referenceId },
       relations: ['author', 'source'],
     });
-    if (!post || !post?.sharedPostId) {
+    if (!post) {
       return;
     }
     const [author, source, sharedPost] = await Promise.all([
       post.author,
       post.source,
-      con.getRepository(Post).findOneBy({ id: post.sharedPostId }),
+      post.type === PostType.Share
+        ? con.getRepository(Post).findOneBy({
+            id: (post as SharePost)?.sharedPostId,
+          })
+        : null,
     ]);
-    if (!author || !source || !sharedPost) {
+    if (!author || !source) {
       return;
     }
-    return {
+
+    const baseObject = {
       full_name: author.name,
-      user_reputation: author.reputation,
       profile_image: author.image,
       squad_name: source.name,
       squad_image: source.image,
-      commentary: truncatePostToTweet(post),
       post_link: addNotificationEmailUtm(
         notification.targetUrl,
         notification.type,
       ),
+      user_reputation: author.reputation,
+    };
+
+    if (!sharedPost) {
+      // Freeform post
+      return {
```

Comment:
Is this right? Regular posts also don't have shared post?

Reply:
What do you consider a regular post? just article post?
You can't be mentioned on a regular post so it won't come to this part.

Reply:
By mentioned you mean in post's content?